spring-observability-core/src/main/java/org/springframework/observability/tracing/listener/HttpServerTracingRecordingListener.java

```diff
@@ -0,0 +1,146 @@
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.observability.tracing.listener;
+
+import org.springframework.observability.core.http.HttpServerRequest;
+import org.springframework.observability.core.http.HttpServerResponse;
+import org.springframework.observability.event.Recording;
+import org.springframework.observability.event.instant.InstantRecording;
+import org.springframework.observability.event.interval.IntervalEvent;
+import org.springframework.observability.event.interval.IntervalHttpServerEvent;
+import org.springframework.observability.event.interval.IntervalRecording;
+import org.springframework.observability.event.listener.RecordingListener;
+import org.springframework.observability.lang.NonNull;
+import org.springframework.observability.tracing.CurrentTraceContext;
+import org.springframework.observability.tracing.Span;
+import org.springframework.observability.tracing.Tracer;
+import org.springframework.observability.tracing.http.HttpServerHandler;
+
+/**
+ * {@link RecordingListener} that uses the Tracing API to record events for HTTP server
+ * side.
+ *
+ * @author Marcin Grzejszczak
+ * @since 1.0.0
+ */
+public class HttpServerTracingRecordingListener extends
+		HttpTracingRecordingListener<HttpServerTracingRecordingListener.TracingContext, HttpServerRequest, HttpServerResponse>
+		implements TracingRecordingListener<HttpServerTracingRecordingListener.TracingContext> {
+
+	private final TracingInstantRecorder tracingInstantRecorder;
+
+	/**
+	 * @param tracer tracer
+	 * @param currentTraceContext current trace context
+	 * @param handler http server handler
+	 */
+	public HttpServerTracingRecordingListener(Tracer tracer, CurrentTraceContext currentTraceContext,
+			HttpServerHandler handler) {
+		super(currentTraceContext, handler::handleReceive, handler::handleSend);
+		this.tracingInstantRecorder = new TracingInstantRecorder(tracer);
+	}
+
+	@Override
+	public boolean isApplicable(Recording<?, ?> recording) {
+		return recording.getEvent() instanceof IntervalHttpServerEvent;
+	}
+
+	@Override
+	public void onStart(IntervalRecording<TracingContext> intervalRecording) {
+		doOnStart(intervalRecording);
+	}
+
+	@Override
+	public void onStop(IntervalRecording<TracingContext> intervalRecording) {
+		doOnStop(intervalRecording);
+	}
+
+	@Override
+	public void onError(IntervalRecording<TracingContext> intervalRecording) {
+
+	}
+
+	@Override
+	public void record(InstantRecording instantRecording) {
+		this.tracingInstantRecorder.record(instantRecording);
```

Comment:
Will this ever happen if this listener is only applicable for `IntervalHttpServerEvent`?

Reply:
If the user wants to record an instantanous event - I can imagine such a case.

Reply:
OK I understood what you're saying - there's no chance this can happen